File: mvc1/Python/10610 30.py
num = input(); sum = 0; answer = 0
num = sorted(num, reverse = True); num = "".join(num)
if num[-1] != '0' : answer = -1
else :
    for i in num :
        sum += int(i)
    answer = -1 if sum%3 != 0 else num
print(answer)
    

File: _etc/C#/10610 30.cs
using System;
using static System.Console;

class Program
{
    static void Main() {
        string s = ReadLine();
        char[] temp = s.ToCharArray();
        Array.Sort(temp); Array.Reverse(temp);
        
        if(temp[temp.Length-1] == '0') {
            int sum = 0;
            for(int i = 0 ; i < temp.Length-1; i++) {
                sum += temp[i] - '0';
            }
            
            if(sum % 3 == 0)  {
                WriteLine(string.Join("", temp));
            } else {
                WriteLine("-1");
            }
        }
        else {
            WriteLine("-1");
        }
    }
}
